docs/desktop-integration.rst

```diff
@@ -144,48 +144,121 @@ it will work as expected from within a sandboxed application.
+``org.kde.WaylandDecoration.QGnomePlatform-decoration`` needs to be
+installed.
 
 Applying themes
 ^^^^^^^^^^^^^^^
 
-There is no ideal way to specify the theme Flatpak applications use. The applications will try to match the system theme currently being used, if it corresponds to any of the Flatpak themes installed, and will fall back to Adwaita (if they use Gtk2 or Gtk3) or the default Qt theme (if they use Qt) if a corresponding theme isn't detected.
+The pre-requisite to applying themes in Flatpaks is to have the theme
+available to the application in the sandbox, commonly done by packaging
+them as theme extensions.
+
+Now the system theme needs to be communicated from host to the sandbox.
+
+On X11, Gtk3 picks up themes from XSettings. Specifically, on GNOME, the
+GNOME XSettings daemon ``gsd-xsettings`` reads the DConf values and
+converts them into XSettings values. On GNOME, this should work by default
+provided ``gsd-xsettings`` is running.
+
+On Wayland, the active theme is exposed via the corresponding Settings
+portal. This requires the portal backend for the desktop environment to
+be installed.
+
+Once the theme is installed and exposed in the sandbox, depending on the
+toolkit/libraries used or the application, it will automatically be
+applied.
+
+If the theme is not available via Flatpak extensions or portal
+or xsettings support is lacking, Gtk and Qt applications will fallback
+to using Adwaita or the default Qt theme.
+
+In this case, either the theme needs to be packaged as an extension
+or the application needs to have permission to read the theme and Gtk or
+Qt settings from host, usually by giving it filesystem access. This
+is undesirable in most cases as it weakens the sandbox and reduces
+portability. The desktop environments should provide proper portal or
+XSettings daemon support.
+
+Appearance Settings
+^^^^^^^^^^^^^^^^^^^
+
+Appearance settings such as the Freedesktop color-scheme preference are
+also exposed similarly via the respective Settings portal. The
+application needs to support reading it and the proper portal backends
+are needed to be installed for this to work.
+
+Icons
+^^^^^
 
-As of Flatpak 0.10.1, the Flatpak system can detect whether the system themes available correspond to any Flatpak themes available in the repositories, and, if so, will automatically install found themes at update time based upon the ``gtk-theme`` Dconf key. This key however can contain only one value, the one of the currently being used theme, which means that the Flatpak versions of matching themes that aren't currently being used aren't installed until those themes are enabled. If none of the corresponding system themes are currently being used, the applications will fall back to Adwaita or the default Qt theme.
+Since Flatpak 0.8.8, host icons from ``/usr/share/icons`` are exposed
+in the sandbox in ``/run/host/share/icons`` and ``$XDG_DATA_HOME/icons``
+in ``/run/host/user-share/icons``.
```

Comment:
We can comment that users should stop putting icons in `~/.icons` also.

Reply:
Added